### DISCUSSION TITLE
Slower sampling speed for LAN likelihood model

### DISCUSSION THREAD

**Discussion by zaizibai** [lnccbrown/HSSM#438] [General]
Dear HSSM developers,

I am trying to use the angel collapse boundary model with HSSM. However, i found MCMC sampling time is much slower than the standard DDM with analytical solution (I have 40 subjects, each finished 160 trials). Since LAN likelihood does not need to compute the infinite series in DDM analytical likelihood, it is reasonable to speculate LAN is faster than DDM analytical likelihood. 

Best,

**Comment by frankmj** [comment]
Did you try this with GPU?
Michael

**Reply by zaizibai** [reply]
Thanks for the response. No, i don't have a Nvidia GPU computer.

**Reply by hcp4715** [reply]
it's been a while for this post, just wondering will GPU make LAN-based hierarchical model faster?

**Comment by digicosmos86** [comment]
Did you use the `nuts_numpyro` sampler?

**Reply by zaizibai** [reply]
I also tried the nuts_numpyro sampler, it did not speed too much.

**Comment by frankmj** [comment]
Yes absolutely! There have also been other improvements that should make
sampling faster but GPUs will definitely speed it up as long as the chains
are not pathological
